index.js

```diff
@@ -0,0 +1,23 @@
+import express from 'express';
+import usersRouter from './routes/users.routes.js';
+import productsRouter from "./routes/products.routes.js";
+import cartsRouter from "./routes/carts.routes.js";
+
+const app = express();
+const port = process.env.PORT || 3001;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
```

Comment:
why is the `express.urlencoded` middleware needed?

Reply:
This option (middleware) is for auto-parsing req.body. From express.js docs - This is a built-in middleware function in Express. It parses incoming requests with urlencoded payloads and is based on body-parser. https://expressjs.com/en/5x/api.html#express.urlencoded:~:text=express.urlencoded(%5Boptions%5D)